lib/logstash/codecs/json.rb

```diff
@@ -44,42 +65,15 @@ def encode(event)
 
   private
 
-  def from_json_parse(json, &block)
-    LogStash::Event.from_json(json).each { |event| yield event }
-  rescue LogStash::Json::ParserError => e
-    @logger.error("JSON parse error, original data now in message field", :error => e, :data => json)
-    yield LogStash::Event.new("message" => json, "tags" => ["_jsonparsefailure"])
+  def parse_json(json)
+    events_from_json(json, targeted_event_factory).each { |event| yield event }
```

Comment:
Can we reliably set an `[event][original]`? Perhaps only if we parse exactly one event?